pysos.py

```diff
@@ -394,35 +359,41 @@ def append(self, value):
         self._indexes.append(key)
         
     def __delitem__(self, i):
-        # trigger observers
-        if self._observers:
-            old_value = self[i]
-            for callback in self._observers:
-                callback(i, None, old_value)
-                
+        self._trigger_observers(i, None, self[i])
         key = self._indexes[i]
         del self._dict[key]
         del self._indexes[i]
     
     def __len__(self):
         return len(self._indexes)
         
-    def __contains__(self, key):
-        raise Exception('Operation not supported for lists')
-    
+    def __contains__(self, value):
+        return value in self._dict.values()
+
+    def insert(self, i, value):
```

Comment:
Funny... I'd explicitly write an explanation comment that "inserting" only works in the front of the list. Moreover, inserting anywhere in-between would be bad performance wise since it would trigger a lot of I/O.

Reply:
Where should I put that comment? Here or to the readme?

Reply:
It's ok, I'll add a comment later